cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/DBSource.java

```diff
@@ -124,6 +131,48 @@ private String getJDBCPluginId() {
     return String.format("%s.%s.%s", "source", dbSourceConfig.jdbcPluginType, dbSourceConfig.jdbcPluginName);
   }
 
+  private void ensureValidConnection(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("could not ensure JDBCDriver is available", e);
+    }
+    try (Connection connection = createConnection()
+    ) {
+      assert(connection.isValid(0));
+    } catch (SQLException e) {
+      throw new IllegalArgumentException("SQL Exception thrown when trying to connect to driver", e);
```

Comment:
Can this method instead just throw ``SQLException``? I feel its better to propagate checked exceptions as far as you can, then convert them into unchecked only if you have to (for example, because the calling method does not allow you to throw an exception)